File: challenges/2026-09-24-longest-balanced-staffing-window/solution.ts
function longestBalancedStaffingWindow(changes: number[]): number {
  // TODO: implement
  return 0;
}

function runSelfChecks(): void {
  const cases: [string, number[], number][] = [
    ["example 1", [3, -3, 4, -2, 2, -4], 6],
    ["example 2", [1, -1, 1, -1, 5], 4],
    ["example 3", [2, 3, -1], 0],
    ["edge: single zero", [0], 1],
    ["edge: no balance possible", [1, 2, 3], 0],
  ];

  for (const [name, changes, expected] of cases) {
    const actual = longestBalancedStaffingWindow(changes);
    const status = actual === expected ? "PASS" : "FAIL";
    console.log(`[${status}] ${name}: expected=${expected} actual=${actual}`);
  }
}

runSelfChecks();

File: challenges/2026-09-24-longest-balanced-staffing-window/solution.py
from typing import List


def longest_balanced_staffing_window(changes: List[int]) -> int:
    """Return the length of the longest contiguous run of `changes`
    that sums to exactly 0."""
    # TODO: implement
    pass


def _run_self_checks() -> None:
    cases = [
        ("example 1", [3, -3, 4, -2, 2, -4], 6),
        ("example 2", [1, -1, 1, -1, 5], 4),
        ("example 3", [2, 3, -1], 0),
        ("edge: single zero", [0], 1),
        ("edge: no balance possible", [1, 2, 3], 0),
    ]

    for name, changes, expected in cases:
        actual = longest_balanced_staffing_window(changes)
        status = "PASS" if actual == expected else "FAIL"
        print(f"[{status}] {name}: expected={expected!r} actual={actual!r}")


if __name__ == "__main__":
    _run_self_checks()
